prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -838,16 +838,15 @@ void rxr_pkt_handle_send_error(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entr
 
 void rxr_pkt_handle_send_completion(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry)
 {
-	struct rdm_peer *peer;
-
-	peer = rxr_ep_get_peer(ep, pkt_entry->addr);
-	if (!peer) {
-		/*
-		 * peer could be NULL in the following 2 scenarios:
-		 * 1. a new peer with same gid+qpn was inserted to av, thus the peer was remove.
-		 * 2. application removed the peer's address from av.
-		 * Either way, we need to ignore this error completion.
-		 */
+	/*
+	 * For a send completion, pkt_entry->addr can be FI_ADDR_NOTAVAIL in 3 situations:
+	 * 1. the pkt_entry is used for a local read operation
+	 * 2. a new peer with same gid+qpn was inserted to av, thus the peer was removed from AV.
+	 * 3. application removed the peer's address from av.
+	 * In 1, we should proceed. For 2 and 3, the send completion should be ignored.
+	 */
+	if (pkt_entry->addr == FI_ADDR_NOTAVAIL &&
```

Comment:
Will this catch all of the failure modes described? I would think that you still want to have a call to `rxr_ep_get_peer` to verify that there is a peer with the address in `pkt_entry->addr`.

Reply:
This function does not use `peer`. For all 3 scenario, `pkt_entry->addr`  will be FI_ADDR_NOTAVAIL